src/lib/operators/multi_predicate_join/multi_predicate_join_evaluator.hpp

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <vector>
+
+#include "operators/operator_join_predicate.hpp"
+#include "storage/segment_accessor.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+enum class PredicateEvaluationResult { False, FalseRightNull, True };
```

Comment:
pls doc (and maybe move to be contained in `MultiPredicateJoinEvaluator`. `PredicateEvaluationResult` is a very general sounding name that is actually only used for MPJ, right)?

Reply:
`PredicateEvaluationResult` as well as `satisfy_all_predicates_detailed_result` is only required to support the currently implemented interpretation of `AntiDiscardNulls` with multiple predicates (https://github.com/hyrise/hyrise/pull/1482#issuecomment-470291988).

Of course we can add additional comments. For me the real question is if the could should remain?

Reply:
"the could"?

> PredicateEvaluationResult as well as satisfy_all_predicates_detailed_result is only required to support the currently implemented interpretation of AntiDiscardNulls with multiple predicates

That's the kind of info I'd like to see doc'ed in code

Reply:
I am sorry: "the could" == "the code"